bluemira/equilibria/solve.py

```diff
@@ -475,7 +475,11 @@ def __init__(
 
     def _optimise_coilset(self):
         try:
-            coilset = self.opt_prob.optimise(fixed_coils=self.fixed_coils)
+            result = self.opt_prob.optimise(fixed_coils=self.fixed_coils)
+            try:
+                coilset = result.coilset
+            except AttributeError:
+                coilset = result
```

Comment:
when would result just be a coilset?

Is this layover from the leftover from the old stuff you mentioned?

Reply:
Yep, sorry, should have left a comment - this just keeps it compatible with the old stuff